metron-platform/metron-common/src/test/java/org/apache/metron/common/field/transformation/RemoveTransformationTest.java

```diff
@@ -0,0 +1,100 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.metron.common.field.transformation;
+
+import com.google.common.collect.Iterables;
+import org.adrianwalker.multilinestring.Multiline;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.apache.metron.common.configuration.FieldTransformer;
+import org.apache.metron.common.configuration.SensorParserConfig;
+import org.json.simple.JSONObject;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.HashMap;
+
+public class RemoveTransformationTest {
+  /**
+   {
+    "fieldTransformations" : [
+          {
+            "input" : "field1"
+          , "transformation" : "REMOVE"
+          }
+                      ]
+   }
+   */
+  @Multiline
+  public static String removeUnconditionalConfig;
+
+  @Test
+  public void testUnconditionalRemove() throws Exception{
+    SensorParserConfig c = SensorParserConfig.fromBytes(Bytes.toBytes(removeUnconditionalConfig));
+    FieldTransformer handler = Iterables.getFirst(c.getFieldTransformations(), null);
+    JSONObject input = new JSONObject(new HashMap<String, Object>() {{
+      put("field1", "foo");
+    }});
+    handler.transformAndUpdate(input, new HashMap<>());
+    Assert.assertFalse(input.containsKey("field1"));
+  }
+
+  /**
+   {
+    "fieldTransformations" : [
+          {
+            "input" : "field1"
+          , "transformation" : "REMOVE"
+          , "config" : {
+              "condition" : "exists(field2) and field2 == 'foo'"
+                       }
+          }
+                      ]
+   }
+   */
+  @Multiline
+  public static String removeConditionalConfig;
+  @Test
+  public void testConditionalRemove() throws Exception {
+    SensorParserConfig c = SensorParserConfig.fromBytes(Bytes.toBytes(removeConditionalConfig));
+    FieldTransformer handler = Iterables.getFirst(c.getFieldTransformations(), null);
+    {
+      JSONObject input = new JSONObject(new HashMap<String, Object>() {{
+        put("field1", "foo");
+      }});
+      handler.transformAndUpdate(input, new HashMap<>());
+      Assert.assertTrue(input.containsKey("field1"));
+    }
+    {
+      JSONObject input = new JSONObject(new HashMap<String, Object>() {{
+        put("field1", "foo");
+        put("field2", "bar");
+      }});
+      handler.transformAndUpdate(input, new HashMap<>());
+      Assert.assertTrue(input.containsKey("field1"));
+    }
+    {
+      JSONObject input = new JSONObject(new HashMap<String, Object>() {{
+        put("field1", "bar");
+        put("field2", "foo");
+      }});
+      handler.transformAndUpdate(input, new HashMap<>());
+      Assert.assertEquals("bar", input.get("field1"));
```

Comment:
Actually, I think you're intending that 'field1' should be removed?  I'm not sure.  Comments would be awesome.  I'm not that bright.